src/Fmt_ast.ml

```diff
@@ -731,9 +731,12 @@ and fmt_core_type c ?(box = true) ?(in_type_declaration = false) ?pro
         fits_breaks
           ( if (protect_token || space_around) && not empty then " ]"
           else "]" )
-          ( if Poly.(c.conf.type_decl = `Sparse) && space_around then
-            "@;<1000 0>]"
-          else "@ ]" )
+          "]"
+          ~hint:
+            ( ( if Poly.(c.conf.type_decl = `Sparse) && space_around then
+                1000
+              else 1 )
+            , 0 )
```

Comment:
Minor style comment: Factorising the `, 0` part make the formating a bit ugly.

Reply:
Done.